src/codegen/doc/format.rs

```diff
@@ -54,121 +55,207 @@ fn get_language<'a>(entry: &'a str, out: &mut String) -> &'a str {
     let out = TAGS.replace_all(&out, "`$0`");
     SPACES.replace_all(&out, " ").into_owned()
 }
+
+fn find_type_by_name(symbol: &str, method_name: Option<&str>, env: &Env) -> String {
+    let symbols = env.symbols.borrow();
+    if let Some(method) = method_name {
+        if let Some((obj_info, fn_info)) = env.analysis.find_object_by_function(
+            env,
+            |o| o.full_name == symbol,
+            |f| f.name == method,
+        ) {
+            let sym = symbols.by_tid(obj_info.type_id).unwrap(); // we are sure the object exists
+            let (type_name, visible_type_name) = obj_info.generate_doc_link_info(fn_info);
+            let name = sym.full_rust_name().replace(&obj_info.name, &type_name);
+
+            fn_info.doc_link(Some(&name), Some(&visible_type_name))
+        } else if let Some((record_info, fn_info)) = env.analysis.find_record_by_function(
+            env,
+            |r| r.type_(&env.library).c_type == symbol,
+            |f| f.name == method,
+        ) {
+            let sym_name = symbols
+                .by_tid(record_info.type_id)
+                .unwrap()
+                .full_rust_name(); // we are sure the object exists
+            fn_info.doc_link(Some(&sym_name), None)
+        } else {
+            format!("`{}::{}`", symbol, method)
+        }
+    } else {
+        find_type(symbol, env)
+    }
+}
+
+fn find_constant_or_variant(symbol: &str, env: &Env) -> String {
+    let symbols = env.symbols.borrow();
+    if let Some(const_info) = env
+        .analysis
+        .constants
+        .iter()
+        .find(|c| c.glib_name == symbol)
+    {
+        // for whatever reason constants are not part of the symbols list
+        format!("[`{name}`][crate::{name}]", name = const_info.name)
+    } else if let Some((flag_info, member_info)) = env.analysis.flags.iter().find_map(|f| {
+        f.type_(&env.library)
+            .members
+            .iter()
+            .find(|m| m.c_identifier == symbol)
+            .map(|m| (f, m))
+    }) {
+        let sym = symbols.by_tid(flag_info.type_id).unwrap();
+        format!(
+            "[`{flag_name}::{member_name}`][crate::{parent}::{member_name}]",
+            member_name = nameutil::bitfield_member_name(&member_info.name),
+            flag_name = flag_info.name,
+            parent = sym.full_rust_name()
+        )
+    } else if let Some((enum_info, member_info)) = env.analysis.enumerations.iter().find_map(|e| {
+        e.type_(&env.library)
+            .members
+            .iter()
+            .find(|m| m.c_identifier == symbol)
+            .map(|m| (e, m))
+    }) {
+        let sym = symbols.by_tid(enum_info.type_id).unwrap();
+        format!(
+            "[`{enum_name}::{member}`][crate::{parent}::{member}]",
+            enum_name = enum_info.name,
+            member = nameutil::enum_member_name(&member_info.name),
+            parent = sym.full_rust_name()
+        )
+    } else {
+        format!("`{}`", symbol)
```

Comment:
this one is not great because now links such as `https://drafts.csswg.org/css-backgrounds-3/#border-radius` will be broken. :/ 
cc @MarijnS95

Reply:
I am not sure what this is about? Theoretically this regex only parses `#Foo`, `%SOME_CONSTANT`, `@parameter-or-self-member`, and `some_function()`, and does not deal with links? Do you have documentation source for an example? I don't recognize the link ;)

Reply:
The link comes from gtk4

Reply:
I need a little more context than that :)

Reply:
```
gtk3-rs/gir-files/Gsk-4.0.gir
2871-The algorithm used for normalizing corner sizes is described in
2872:[the CSS specification](https://drafts.csswg.org/css-backgrounds-3/#border-radius).</doc>
2873-      <field name="bounds" writable="1">

gtk-rs-core/gir-files/Gsk-4.0.gir
2871-The algorithm used for normalizing corner sizes is described in
2872:[the CSS specification](https://drafts.csswg.org/css-backgrounds-3/#border-radius).</doc>
2873-      <field name="bounds" writable="1">

gtk4-rs/gir-files/Gsk-4.0.gir
2871-The algorithm used for normalizing corner sizes is described in
2872:[the CSS specification](https://drafts.csswg.org/css-backgrounds-3/#border-radius).</doc>
2873-      <field name="bounds" writable="1">
```

Those are normal markdown links, and should not be touched by this code IMO.

Reply:
Sorry I'm on a phone right now that's why I couldn't provide more context. The regex still catches the link and breaks it by surrounding it with ``

Reply:
I guess we should prevent parsing things inside `[]()` instead... Ugh :)

Reply:
without look-ahead/look-behind? :p